tachyon/zk/plonk/halo2/snark_verifier_poseidon_transcript.h

```diff
@@ -55,7 +56,7 @@ class SnarkVerifierPoseidonBase {
   }
 
   ScalarField DoSqueeze() {
-    size_t rate = poseidon_.config.rate;
+    constexpr size_t rate = Params::kRate;
```

Comment:
Could you replace `poseidon_.config.rate` with `Params::kRate` explicitly like you did in `tachyon/zk/plonk/halo2/poseidon_transcript.h`?